test/jdk/java/security/Provider/ChangeProviders.java

```diff
@@ -81,6 +81,22 @@ public static void main0(String[] args) throws Exception {
             throw new Exception("Provider not at pos 1");
         }
 
+        //Ensure that providers inserted at position 0 or > n are placed
+        //at the n+1st position
```

Comment:
It's probably clearer to reset n after this and then test for n+1 instead of n+2, ex - add a line:

```
// reset n to number of installed providers
n = plen();
```